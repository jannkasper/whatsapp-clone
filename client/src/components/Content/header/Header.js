import React from "react";
import { Menu, Search } from "../../icons"
import styles from "./header.module.scss";
import defaultAvatar from "../../../img/avatar.svg";

function Header({ currentContact }) {
    let baseImage = defaultAvatar;
    if (currentContact.profileImage) {
        baseImage = `data:${currentContact.profileImage.type};base64,${currentContact.profileImage.data}`
    }

    return (
        <div className={styles.headerContainer}>
            <div className={styles.userAvatar}>
                <img className={styles.image} src={baseImage} alt="profileImage"/>
            </div>
            <div className={styles.userName}>
                <span>{currentContact.username}</span>
            </div>
            <div className={styles.iconGroup}>
                <div className={styles.icon}>
                    <Search />
                </div>
                <div className={styles.icon}>
                    <Menu />
                </div>
            </div>
        </div>
    )
}

export default Header
