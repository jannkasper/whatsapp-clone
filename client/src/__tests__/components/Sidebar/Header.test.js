import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from '../../../store';
import Header from '../../../components/Sidebar/header';

const store = configureStore();

describe('Header component', () => {

    it('should display', () => {
        const {asFragment} = render(
            <Provider store={store}>
                <Header />
            </Provider>,
        );

        expect(asFragment()).toMatchSnapshot();
    })
});
