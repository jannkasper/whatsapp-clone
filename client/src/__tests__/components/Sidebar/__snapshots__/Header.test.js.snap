// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conversation component should display 1`] = `
<DocumentFragment>
  <div
    class="headerContainer"
  >
    <div
      class="userImage"
    >
      <img
        alt="profileImage"
        class="image"
        src="avatar.svg"
      />
    </div>
    <div
      class="iconGroup"
    >
      <div
        class="icon"
      >
        <span
          class=""
          data-icon="status-v3-unread"
          data-testid="status-v3-unread"
        >
          <svg
            height="24"
            id="df9d3429-f0ef-48b5-b5eb-f9d27b2deba6"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.072 1.761a10.05 10.05 0 0 0-9.303 5.65.977.977 0 0 0 1.756.855 8.098 8.098 0 0 1 7.496-4.553.977.977 0 1 0 .051-1.952zM1.926 13.64a10.052 10.052 0 0 0 7.461 7.925.977.977 0 0 0 .471-1.895 8.097 8.097 0 0 1-6.012-6.386.977.977 0 0 0-1.92.356zm13.729 7.454a10.053 10.053 0 0 0 6.201-8.946.976.976 0 1 0-1.951-.081v.014a8.097 8.097 0 0 1-4.997 7.209.977.977 0 0 0 .727 1.813l.02-.009z"
              fill="currentColor"
            />
            <path
              d="M19 1.5a3 3 0 1 1 0 6 3 3 0 0 1 0-6z"
              fill="#009588"
            />
          </svg>
        </span>
      </div>
      <div
        class="icon"
      >
        <span
          class=""
          data-icon="chat"
          data-testid="chat"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.005 3.175H4.674C3.642 3.175 3 3.789 3 4.821V21.02l3.544-3.514h12.461c1.033 0 2.064-1.06 2.064-2.093V4.821c-.001-1.032-1.032-1.646-2.064-1.646zm-4.989 9.869H7.041V11.1h6.975v1.944zm3-4H7.041V7.1h9.975v1.944z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="icon"
      >
        <span
          class=""
          data-icon="menu"
          data-testid="menu"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 7a2 2 0 1 0-.001-4.001A2 2 0 0 0 12 7zm0 2a2 2 0 1 0-.001 3.999A2 2 0 0 0 12 9zm0 6a2 2 0 1 0-.001 3.999A2 2 0 0 0 12 15z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Header component should display 1`] = `
<DocumentFragment>
  <div
    class="headerContainer"
  >
    <div
      class="userImage"
    >
      <img
        alt="profileImage"
        class="image"
        src="avatar.svg"
      />
    </div>
    <div
      class="iconGroup"
    >
      <div
        class="icon"
      >
        <span
          class=""
          data-icon="status-v3-unread"
          data-testid="status-v3-unread"
        >
          <svg
            height="24"
            id="df9d3429-f0ef-48b5-b5eb-f9d27b2deba6"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.072 1.761a10.05 10.05 0 0 0-9.303 5.65.977.977 0 0 0 1.756.855 8.098 8.098 0 0 1 7.496-4.553.977.977 0 1 0 .051-1.952zM1.926 13.64a10.052 10.052 0 0 0 7.461 7.925.977.977 0 0 0 .471-1.895 8.097 8.097 0 0 1-6.012-6.386.977.977 0 0 0-1.92.356zm13.729 7.454a10.053 10.053 0 0 0 6.201-8.946.976.976 0 1 0-1.951-.081v.014a8.097 8.097 0 0 1-4.997 7.209.977.977 0 0 0 .727 1.813l.02-.009z"
              fill="currentColor"
            />
            <path
              d="M19 1.5a3 3 0 1 1 0 6 3 3 0 0 1 0-6z"
              fill="#009588"
            />
          </svg>
        </span>
      </div>
      <div
        class="icon"
      >
        <span
          class=""
          data-icon="chat"
          data-testid="chat"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.005 3.175H4.674C3.642 3.175 3 3.789 3 4.821V21.02l3.544-3.514h12.461c1.033 0 2.064-1.06 2.064-2.093V4.821c-.001-1.032-1.032-1.646-2.064-1.646zm-4.989 9.869H7.041V11.1h6.975v1.944zm3-4H7.041V7.1h9.975v1.944z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="icon"
      >
        <span
          class=""
          data-icon="menu"
          data-testid="menu"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 7a2 2 0 1 0-.001-4.001A2 2 0 0 0 12 7zm0 2a2 2 0 1 0-.001 3.999A2 2 0 0 0 12 9zm0 6a2 2 0 1 0-.001 3.999A2 2 0 0 0 12 15z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
